packages/overlay/src/OverlayTrigger.ts

```diff
@@ -27,29 +27,59 @@ import type { Placement } from '@floating-ui/dom';
 import type { BeforetoggleOpenEvent } from './events.js';
 import type { Overlay } from './Overlay.js';
 import type { OverlayTriggerInteractions } from './overlay-types';
+import '@spectrum-web-components/overlay/sp-overlay.js';
 
 import overlayTriggerStyles from './overlay-trigger.css.js';
 
 export type OverlayContentTypes = 'click' | 'hover' | 'longpress';
 
+// Helper type to create all unique combinations of OverlayContentTypes
+type Combinations<T extends string, U extends string = T> = T extends string
+    ? T | `${T} ${Combinations<Exclude<U, T>>}`
+    : never;
+
+export type TriggeredByType = Combinations<OverlayContentTypes>;
+
 /**
  * @element overlay-trigger
  *
+ * A component that manages overlay content triggered by different interactions.
+ * Supports click, hover, and longpress triggered overlays with configurable
+ * placement and behavior.
+ *
  * @slot trigger - The content that will trigger the various overlays
  * @slot hover-content - The content that will be displayed on hover
  * @slot click-content - The content that will be displayed on click
- * @slot longpress-content - The content that will be displayed on click
+ * @slot longpress-content - The content that will be displayed on longpress
+ * @slot longpress-describedby-descriptor - Description for longpress content
  *
  * @fires sp-opened - Announces that the overlay has been opened
  * @fires sp-closed - Announces that the overlay has been closed
+ *
+ * @attr {string} placement - The placement of the overlay relative to the trigger
+ * @attr {number} offset - The distance between the overlay and the trigger
+ * @attr {boolean} disabled - Whether the overlay trigger is disabled
+ * @attr {string} receives-focus - How focus should be handled ('true'|'false'|'auto')
  */
 export class OverlayTrigger extends SpectrumElement {
     public static override get styles(): CSSResultArray {
         return [overlayTriggerStyles];
     }
 
-    @property()
-    content = 'click hover longpress';
+    /**
+     * Optional property to optimize performance and prevent race conditions.
+     *
+     * By explicitly declaring which content types are used (e.g. "click", "longpress hover"),
+     * we can avoid:
+     * 1. Extra renders from unnecessary slot reparenting
+     * 2. Potential infinite render loops during content detection
+     * 3. Race conditions during slot assignment
+     *
+     * By only returning overlay wrappers for explicitly declared content types,
+     * we minimize unecessary DOM nodes, operations and ensure a more stable rendering behavior.
+     */
+    @property({ attribute: 'triggered-by' })
+    triggeredBy?: TriggeredByType;
```

Comment:
This will not surface up in the API tab. You need to make it public!

Reply:
fixed!